<div class="container">

<table style="width: 100%;"><tr>
<td>lkmt-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>An S4 class to represent the variance component test.</h2>

<h3>Description</h3>

<p>An S4 class to represent the variance component test.
</p>
<p><code>show</code> displays basic information on <code>lkmt</code> object
</p>
<p><code>summary</code> generates a <code>lkmt</code> object summary including the used kernel, pathway and the test result
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S4 method for signature 'lkmt'
show(object)

## S4 method for signature 'lkmt'
summary(object)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>An object of class <code>lkmt</code>.</p>
</td>
</tr></table>
<h3>Value</h3>

<p><code>show</code> Basic information on <code>lkmt</code> object.
</p>
<p><code>summary</code> Summarized information on <code>lkmt</code> object.
</p>


<h3>Slots</h3>


<dl>
<dt><code>formula</code></dt>
<dd>
<p>A formula stating the regression nullmodel that will be used in
the variance component test.</p>
</dd>
<dt><code>kernel</code></dt>
<dd>
<p>An object of class <code>kernel</code> representing the similarity
matrix of the individuals based on which the pathways influence is evaluated.</p>
</dd>
<dt><code>GWASdata</code></dt>
<dd>
<p>An object of class <code>GWASdata</code> including the data
on which the test is conducted.</p>
</dd>
<dt><code>statistic</code></dt>
<dd>
<p>A <code>vector</code> giving the value of the variance component
test statistic.</p>
</dd>
<dt><code>df</code></dt>
<dd>
<p>A <code>vector</code> containing the number of degrees of freedom.</p>
</dd>
<dt><code>p.value</code></dt>
<dd>
<p>A <code>vector</code> giving the p-value calculated for the <code>pathway</code> object considered in the variance component test.
</p>
<p>For details on the variance component test see the references.</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Juliane Manitz, Stefanie Friedrichs
</p>


<h3>References</h3>


<ul>
<li>
<p> Liu D, Lin X, Ghosh D: Semiparametric regression of multidimensional genetic pathway data: least-squares kernel machines and linear mixed models. Biometrics 2007, 63(4):1079-88.
</p>
</li>
<li>
<p> Wu MC, Kraft P, Epstein MP, Taylor DM, Chanock SJ, Hunter DJ, Lin X: Powerful SNP-Set Analysis for Case-Control Genome-Wide Association Studies. Am J Hum Genet 2010, 86:929-42
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">data(hsa04020)
data(gwas)
# compute kernel
net_kernel &lt;- calc_kernel(gwas, hsa04020, knots=NULL, type='net', calculation='cpu')
# perform LKMT 
res &lt;- lkmt_test(pheno ~ sex + age, net_kernel, gwas, method='satt')
# show and summary methods
show(res)
summary(res)
# summary method
summary(lkmt.net.kernel.hsa04020)
</code></pre>


</div>