<div class="container">

<table style="width: 100%;"><tr>
<td>countBarcodedReads</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get read counts from barcoded reads</h2>

<h3>Description</h3>

<p><code>countBarcodedReads()</code> and <code>countBarcodedReadsFromContig()</code> match exogenous DNA barcode sequences to their associated
cell barcodes and saves them to the <code>colData</code> (cell barcode metadata) of <code>TapestriExperiment</code>.
<code>countBarcodedReads()</code> is a shortcut for <code>countBarcodedReadsFromContig()</code>, allowing the user to specify 'gRNA' or 'barcode'
to use the <code>grnaCounts</code> or <code>barcodeCounts</code> <code>altExp</code> slots.
The entries in the <code>barcode.lookup</code> table do not have to be present in the sample,
allowing users to keep one master table/file of available barcode sequences for use in all experiments.
The <code>Rsamtools</code> and <code>Biostrings</code> packages must be installed to use these functions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">countBarcodedReads(
  TapestriExperiment,
  bam.file,
  barcode.lookup,
  probe,
  return.table = FALSE,
  max.mismatch = 2,
  with.indels = FALSE,
  ...
)

countBarcodedReadsFromContig(
  bam.file,
  barcode.lookup,
  contig,
  cell.barcode.tag = "RG",
  max.mismatch = 2,
  with.indels = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>TapestriExperiment</code></td>
<td>
<p><code>TapestriExperiment</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bam.file</code></td>
<td>
<p>File path of BAM file. <code>.bai</code> BAM index file must be in the same location (can be generated using <code>Rsamtools::indexBam()</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>barcode.lookup</code></td>
<td>
<p><code>data.frame</code> where the first column is the barcode identifier/name and the second column is the DNA sequence. Headers are ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>probe</code></td>
<td>
<p>Character, either "gRNA" or "barcode" to parse counts from <code>grnaCounts</code> or <code>barcodeCounts</code> <code>altExp</code> slots, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.table</code></td>
<td>
<p>Logical, if <code>TRUE</code>, returns table of read counts per barcode. If <code>FALSE</code>, returns <code>TapestriExperiment.</code> Default <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.mismatch</code></td>
<td>
<p>Numeric, the maximum and minimum number of mismatching letters allowed. Default 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>with.indels</code></td>
<td>
<p>If <code>TRUE</code>, then indels are allowed. Default <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Arguments to pass on to <code>countBarcodedReadsFromContig()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contig</code></td>
<td>
<p>Character, contig or chromosome name to search for barcodes in. Can be a vector of more than one contig to expand search space.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cell.barcode.tag</code></td>
<td>
<p>Character of length 2, indicates cell barcode field in BAM, specified by Tapestri pipeline (currently "RG"). Default "RG".</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>TapestriExperiment</code> with barcoded read counts added to <code>colData</code>.
</p>
<p>A data.frame of read counts for each specified barcode.
</p>


<h3>See Also</h3>

<p><code>Rsamtools::indexBam()</code>
</p>
<p><code>Biostrings::matchPattern()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
counts &lt;- countBarcodedReads(
  TapestriExperiment,
  bam.file, barcode.lookup, "gRNA"
)

## End(Not run)
## Not run: 
counts &lt;- countBarcodedReadsFromContig(bam.file, barcode.lookup, "virus_ref2")

## End(Not run)
</code></pre>


</div>