<div class="container">

<table style="width: 100%;"><tr>
<td>kc.marginal</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Marginal Maximum Likelihood estimation of kin-cohort data </h2>

<h3>Description</h3>

<p>This function estimates cumulative risk and hazard at given ages for carriers and noncarriers of a mutation based on the probands genotypes. It uses the Marginal Maximum Likelihood estimation method (Chatterjee and Wacholder, 2001). Piece-wise exponential distribution is assumed for the survival function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">kc.marginal(t, delta, genes, r, knots, f, pw = rep(1,length(t)), 
            set = NULL, B = 1, maxit = 1000, tol = 1e-5, subset,
            logrank=TRUE, trace=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>t</code></td>
<td>
<p> time variable. Usually age at diagnosis or at last follow-up </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delta</code></td>
<td>
<p> disease status (1: event, 0: no event </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p> factor or numeric vector (1 gene), matrix or dataframe (2 genes) with genotypes of proband numeric. factors and data.frame with factors are prefered in order to use user-defined labels. Otherwise use codes (1:noncarrier, 2: carrier, 3: homozygous carrier) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r</code></td>
<td>
<p> relationship with proband 1:parent, 2:sibling 3:offspring 0:proband. Probands will be excluded from analysis and offspring will be recoded 1 internally.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>knots</code></td>
<td>
<p> time points (ages) for cumulative risk and hazard estimates </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>f</code></td>
<td>
<p> vector of mutation allele frequencies in the population </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pw</code></td>
<td>
<p> prior weights, if needed </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>set</code></td>
<td>
<p> family id (only needed for bootstrap)   </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>
<p> number of boostrap samples (only needed for bootstrap) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxit</code></td>
<td>
<p> max number of iterations for the EM algorithm </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>
<p> convergence tolerance </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subset</code></td>
<td>
<p> logical condition to subset data </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logrank</code></td>
<td>
<p> Perform a logrank test </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trace</code></td>
<td>
<p> Show iterations for bootstrap </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>object of classes "kin.cohort" and "chatterjee". 
</p>
<table>
<tr style="vertical-align: top;">
<td><code>cumrisk </code></td>
<td>
<p>matrix with cumulative risk estimates for noncarriers, carriers and the cumulative risk ratio. Estimates are given for the times indicated in the knot vector</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hazard </code></td>
<td>
<p>matrix with hazard estimates for noncarriers, carriers and the hazard ratio. Estimates are given for the times indicated in the knot vector</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>knots</code></td>
<td>
<p>vector of knots</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conv</code></td>
<td>
<p>if the EM algorithm converged</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>niter</code></td>
<td>
<p>number of iterations needed for convergence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ngeno.rel</code></td>
<td>
<p>number of combinations of genotypes in the relatives</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>events</code></td>
<td>
<p>matrix with number of events and person years per each knot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logHR</code></td>
<td>
<p>mean log hazard ratio estimate (unweighted)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logrank</code></td>
<td>
<p>logrank test. If 2 genes, for the main effects, the cross-classification and the stratified tests</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p>copy of call</p>
</td>
</tr>
</table>
<p>if bootstrap confidence intervals are requested (B&gt;1) then the returned object is of classes "kin.cohort.boot" and "chatterjee" with previous items packed in value estimate and each bootstrap sample packed in matrices. 
</p>


<h3>Note</h3>

 
<p>This function is best called by kin.cohort than directly
</p>


<h3>References</h3>

 
<p>Chatterjee N and Wacholder S. A Marginal Likelihood Approach for Estimating Penetrance from Kin-Cohort Designs. Biometrics. 2001; 57: 245-52. 
</p>


<h3>See Also</h3>

 
<p><code>kin.cohort</code>, <code>print.kin.cohort</code>, <code>plot.kin.cohort</code> 
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
data(kin.data)
attach(kin.data)
res.mml&lt;- kc.marginal(age, cancer, gen1, rel, knots=c(30,40,50,60,70,80), f=0.02)
res.mml

## End(Not run)
</code></pre>


</div>