<div class="container">

<table style="width: 100%;"><tr>
<td>rpt.IBD</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Replacing CPTs for selected mixture contributors with familial relationships</h2>

<h3>Description</h3>

<p>Used after a call to <code>DNAmixture</code> with <code>compile=FALSE,triangulate=FALSE</code>, this function replaces the CPTs for the genotype allele count arrays for specified mixture contributors by those representing the specified relationship with each other and typed relatives
</p>


<h3>Usage</h3>

<pre><code class="language-R">rpt.IBD(mixture, IBD="parent-child", typed.gts = NULL, inds = 1, 
jtyped = ncol(IBD$patt)/2 - length(typed.gts) + seq_along(typed.gts), 
jcontr = seq_along(inds), 
targets=attr(IBD,'targets'), contribs=NULL, quiet=FALSE, all.freq = NULL, compile = TRUE) 

rpt.typed.relatives(mixture, IBD="parent-child", typed.gts = NULL, inds = 1, 
jtyped = ncol(IBD$patt)/2 - length(typed.gts) + seq_along(typed.gts), 
jcontr = seq_along(inds), 
targets=attr(IBD,'targets'), contribs=NULL, quiet=FALSE, all.freq = NULL, compile = TRUE)

rpt.typed.child(mixture, aca, ind=1)

replace.Ui.tables(mixture, aca, ind=1)

rpt.typed.parents(mixture, Mgt, Fgt, ind=1, compile=TRUE)

rpt.typed.relative(mixture, Rgt, IBD=c(0.25,0.5,0.25), ind=1, compile=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mixture</code></td>
<td>
<p>DNAmixtures object created by previous call to <code>DNAmixture</code> with <br><code>triangulate=FALSE,compile=FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IBD</code></td>
<td>
<p>relationships between the specified individuals, as multi-person condensed 
coefficients of identity, in one of several representation; see Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>typed.gts</code></td>
<td>
<p>list of 0 or more genotypes of relatives; the components of this list must
be named (with the id's of the relevant individuals) if <code>targets</code> and <code>contribs</code>
are used to code the correspondences (see Details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inds</code></td>
<td>
<p>vector of 1 or more integers: which ‘unknown’ contributors are we modelling by
amending their CPTs? The elements should be listed in the same order as the corresponding
pairs of columns of the IBD patterns in <code>IBD</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>jtyped</code></td>
<td>
<p>indices of pairs of columns of <code>IBD$patt</code> that correspond to the 
typed relatives (if any); default the last <code>length(typed.gts)</code> pairs of columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>jcontr</code></td>
<td>
<p>indices of pairs of columns of <code>IBD$patt</code> that correspond to the 
relevant mixture contributors; default the first <code>length(inds)</code> pairs of columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>targets</code></td>
<td>
<p>Character vector of the tags of the individuals referred to in <code>IBD</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contribs</code></td>
<td>
<p>Character vector of the tags of the individuals included in the mixture,
in order</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>should calculated values of inds, jtyped and jcontr be reported?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>all.freq</code></td>
<td>
<p>alternative allele frequency database(s), see Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>compile</code></td>
<td>
<p>logical flag: should mixture object be compiled on exit?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind</code></td>
<td>
<p>as <code>inds</code>, used above when only one allowed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aca, Mgt, Fgt, Rgt</code></td>
<td>
<p>individual genotypes, as allele count arrays</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In using <code>rpt.IBD</code> or <code>rpt.typed.relatives</code> (which is identical), the correspondence between mixture contributors, specified relationships, and typed genotype profiles should be specified 
<strong>either</strong> (preferably)
using <code>targets</code>, <code>contribs</code> and through the names of the components in <code>typed.gts</code>,
<strong>or</strong> (to be deprecated)
with <code>inds</code>, <code>jcontr</code> and <code>jtyped</code>: 
the two representations should not be mixed up. If either <code>targets</code> or <code>contribs</code> specified, the former representation is assumed.
</p>
<p>Special cases are treated slightly more efficiently:
<code>rpt.typed.child</code>: single contributor, single typed relative, parent or child;
<code>rpt.typed.parents</code>: single contributor, both parents typed;
<code>rpt.typed.relative</code>: single contributor, single typed relative.
</p>
<p>Note that <code>IBD$patt</code> always has an even number of columns, two for each individual
in the joint relationship specified; <code>jtyped</code> and <code>jcontr</code> are vectors of
indices of these individuals, i.e. to pairs of adjacent columns of <code>IBD$patt</code>.
</p>
<p>Multiple functions in this group can validly be called sequentially (with all but the
last having <code>compile=FALSE</code>) providing they reference different sets of
contributors among the targets, <strong>and</strong> that these sets are conditionally
independent given the typed genotypes specified.
</p>
<p>There are multiple valid representations for relationships in the argument <code>IBD</code> –
as an IBD pattern distribution, via a pedigree, or. in the case of just two individuals.
via either a vector of 3 kappas or 9 Deltas (Jacquard's condensed coefficients of
identity). For full details, see <code>convertIBD</code>.
If <code>IBD</code> is missing, the default value represents parent-child.
</p>
<p>In the interests of upward compatibility, in <code>rpt.typed.child</code> and 
<code>replace.Ui.tables</code> (which are identical), the argument <code>Cgt</code> can be given as
either a genotype profile data frame, or an allele count array.
</p>
<p>By default, the allele frequency database used for the founding genes is that
used when the <code>mixture</code> object is created, in an earlier call to <code>DNAmixture</code>.
A non-null value for the <code>all.freq</code> argument allows the
user to specify alternative database (s) for the founding genes. If its value
is an allele frequency database (in the format specified in 
<code>formats</code>) then that database is used for all founding genes; if the
value of the argument is a list of such databases, then component k of the
list is used for allele frequencies for the founding gene labelled k in the
<code>IBD</code> argument. Note that this option allows modelling of mixtures where
different contributors are drawn from different populations, whether or not
there are relationships among individuals. </p>


<h3>Value</h3>

<p>Vector of marker-specific probabilities of the typed genotypes.</p>


<h3>Author(s)</h3>

<p>Peter Green (P.J.Green@bristol.ac.uk)
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(test2data)
data(NGMDyes)

## Fit 2-person mixture - baseline model

mixD&lt;-DNAmixture(list(epg),k=2,C=list(0.001),database=db)
pars&lt;-mixpar(rho=list(2),eta=list(100),xi=list(0.1),phi=list(c(U1=0.7,U2=0.3)))
baseline&lt;-logL(mixD)(pars)

## Fit 2-person mixture model in which contributor 1 is parent of a typed individual Cgt

mixD&lt;-DNAmixture(list(epg),k=2,C=list(0.001),database=db,triangulate=FALSE,compile=FALSE)
rpt.IBD(mixD,,list(Cgt)) 
log10LR&lt;-(logL(mixD)(pars)-baseline)/log(10)
cat('log10 LR',log10LR,'\n')

## Fit 2-person mixture model in which contributor 1 is father of a typed individual Cgt 
## with mother Mgt

mixD&lt;-DNAmixture(list(epg),k=2,C=list(0.001),database=db,triangulate=FALSE,compile=FALSE)
rpt.IBD(mixD,,list(Mgt,Cgt)) 
log10LR&lt;-(logL(mixD)(pars)-baseline)/log(10)
cat('log10 LR',log10LR,'\n')

## Fit 2-person mixture model in which contributors are two parents of a child with 
## genotype Cgt, and a parent of one of them has genotype Rgt. Note the encoding of allele 
## labels to reduce the complexity of the IBD pattern distribution IBD.

IBD&lt;-list(patt=rbind(c(1,3,2,4,1,2,1,5),c(1,3,2,4,1,2,3,5)))

mixD&lt;-DNAmixture(list(epg),k=2,C=list(0.001),database=db,triangulate=FALSE,compile=FALSE)
rpt.IBD(mixD,IBD,list(Cgt,Rgt),1:2) 
log10LR&lt;-(logL(mixD)(pars)-baseline)/log(10)
cat('log10 LR',log10LR,'\n')

## the same, with individuals and relationships denoted by character tags

mixD&lt;-DNAmixture(list(epg),k=2,C=list(0.001),database=db,triangulate=FALSE,compile=FALSE)
rpt.IBD(mixD,IBD,list(c=Cgt,gf=Rgt),targets=c('f','m','c','gf'),contribs=c('f','m')) 
log10LR&lt;-(logL(mixD)(pars)-baseline)/log(10)
cat('log10 LR',log10LR,'\n')
</code></pre>


</div>