<div class="container">

<table style="width: 100%;"><tr>
<td>kmeRs_similarity_matrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pairwise Similarity Matrix</h2>

<h3>Description</h3>

<p>The <code>kmeRs_similarity_matrix</code> function generates a pairwise similarity score
matrix for for k length given k-mers vs. all possible k-mers combination.
The pairwise similarity score is calculated using PAM or BLOSUM substitution matrix;
30, 40, 70, 120, 250 and 62, 45, 50, 62, 80, 100 matrix versions are available for
PAM or BLOSUM, respectively. The results are evaluated by global similarity score;
higher similarity score indicates more similar sequences for BLOSUM and opposite for
PAM matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">kmeRs_similarity_matrix(
  q = NULL,
  x = NULL,
  align.type = "global",
  k = 3,
  seq.type = "AA",
  submat = ifelse(test = (match.arg(toupper(seq.type), c("DNA", "AA")) == "AA"), yes =
    "BLOSUM62", no = NA),
  compare.all = FALSE,
  save_to_file = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>q</code></td>
<td>
<p>query vector with given k-mers</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>kmers to search the query vector against. If unspecified, <code>q</code> will 
be compared to either other k-mers within <code>q</code> (<code>compare.all = FALSE</code>),
or all possible combinations specified by the parameter <code>k</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>align.type</code></td>
<td>
<p>type of alignment, either <code>global</code> or <code>local</code>.
<code>global</code> uses Needleman-Wunsch global alignment to calculate scores, while
<code>local</code> represents Smith-Waterman local alignment instead</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>length of k-mers to calculate the similarity matrix for, defaults to 3; e.g. for DNA, N = 4^3 = 64 combinations if <code>k = 3</code>;</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq.type</code></td>
<td>
<p>type of sequence in question, either 'DNA' or 'AA' (default);
this will also modify <code>q</code> accordingly, if <code>q</code> is unspecified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>submat</code></td>
<td>
<p>substitution matrix, default to 'BLOSUM62'; other choices are
'BLOSUM45', 'BLOSUM50', 'BLOSUM62', 'BLOSUM80', 'BLOSUM100', 'PAM30',
'PAM40', 'PAM70', 'PAM120' or 'PAM250'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>compare.all</code></td>
<td>
<p>if <code>TRUE</code>, the query vector will be compared to all
possible combinations of k-mers (defaults to <code>FALSE</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save_to_file</code></td>
<td>
<p>if specified, the results will be saved to the path in
comma-separated format (.CSV)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other parameters, e.g. gap opening/extension penalties (<code>gapOpening</code>,
<code>gapExtension</code>), or DNA match/mismatch scores (<code>na.match</code>, <code>na.mismatch</code>)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>similarity matrix is returned as a data.frame
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Simple BLOSUM62 similarity matrix for all amino acid nucleotides
kmeRs_similarity_matrix(submat = "BLOSUM62")

</code></pre>


</div>