<div class="container">

<table style="width: 100%;"><tr>
<td>cluster</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Divisive k-means clustering.</h2>

<h3>Description</h3>

<p>This function recursively splits a sequence set into smaller and smaller subsets,
returning a "dendrogram" object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">cluster(x, k = 5, residues = NULL, gap = "-", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a list or matrix of sequences, possibly an object of class
<code>"DNAbin"</code> or <code>"AAbin"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>integer. The k-mer size required.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>residues</code></td>
<td>
<p>either NULL (default; emitted residues are automatically
detected from the sequences), a case sensitive character vector
specifying the residue alphabet, or one of the character strings
"RNA", "DNA", "AA", "AMINO". Note that the default option can be slow for
large lists of character vectors. Specifying the residue alphabet is therefore
recommended unless the sequence list is a "DNAbin" or "AAbin" object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap</code></td>
<td>
<p>the character used to represent gaps in the alignment matrix
(if applicable). Ignored for <code>"DNAbin"</code> or <code>"AAbin"</code> objects.
Defaults to "-" otherwise.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to <code>kmeans</code> (not including
<code>centers</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function creates a tree by successively splitting
the dataset into smaller and smaller subsets (recursive
partitioning). This is a divisive, or "top-down" approach to tree-building,
as opposed to agglomerative "bottom-up" methods such as neighbor joining
and UPGMA. It is particularly useful for large large datasets with many sequences
(<em>n</em> &gt; 10,000) since the need to compute a large <em>n</em> * <em>n</em>
distance matrix is circumvented.
Instead, a matrix of k-mer counts is computed, and split recursively row-wise
using a k-means clustering algorithm (<em>k</em> = 2). This effectively reduces
the time and memory complexity from quadratic to linear, while generally
maintaining comparable accuracy.
</p>
<p>If a more accurate tree is required, users can increase the value
of <code>nstart</code> passed to <code>kmeans</code> <em>via</em> the <code>...</code> argument.
While this can increase computation time, it can improve tree accuracy
considerably.
</p>
<p>DNA and amino acid sequences can be passed to the function either as
a list of non-aligned sequences or a matrix of aligned sequences,
preferably in the "DNAbin" or "AAbin" raw-byte format
(Paradis et al 2004, 2012; see the <code>ape</code> package
documentation for more information on these S3 classes).
Character sequences are supported; however ambiguity codes may
not be recognized or treated appropriately, since raw ambiguity
codes are counted according to their underlying residue frequencies
(e.g. the 5-mer "ACRGT" would contribute 0.5 to the tally for "ACAGT"
and 0.5 to that of "ACGGT").
</p>
<p>To minimize computation time when counting longer k-mers (k &gt; 3),
amino acid sequences in the raw "AAbin" format are automatically
compressed using the Dayhoff-6 alphabet as detailed in Edgar (2004).
Note that amino acid sequences will not be compressed if they
are supplied as a list of character vectors rather than an "AAbin"
object, in which case the k-mer length should be reduced
(k &lt; 4) to avoid excessive memory use and computation time.
</p>


<h3>Value</h3>

<p>Returns an object of class <code>"dendrogram"</code>.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>References</h3>

<p>Edgar RC (2004) Local homology recognition and distance measures in
linear time using compressed amino acid alphabets.
<em>Nucleic Acids Research</em>, <strong>32</strong>, 380-385.
</p>
<p>Paradis E, Claude J, Strimmer K, (2004) APE: analyses of phylogenetics
and evolution in R language. <em>Bioinformatics</em> <strong>20</strong>, 289-290.
</p>
<p>Paradis E (2012) Analysis of Phylogenetics and Evolution with R
(Second Edition). Springer, New York.
</p>


<h3>See Also</h3>

<p><code>kcount</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
## Cluster the woodmouse dataset (ape package)
library(ape)
data(woodmouse)
## trim gappy ends to subset global alignment
woodmouse &lt;- woodmouse[, apply(woodmouse, 2, function(v) !any(v == 0xf0))]
## build tree divisively
suppressWarnings(RNGversion("3.5.0"))
set.seed(999)
woodmouse.tree &lt;- cluster(woodmouse, nstart = 5)
## plot tree
op &lt;- par(no.readonly = TRUE)
par(mar = c(5, 2, 4, 8) + 0.1)
plot(woodmouse.tree, main = "Woodmouse phylogeny", horiz = TRUE)
par(op)

## End(Not run)
</code></pre>


</div>