<div class="container">

<table style="width: 100%;"><tr>
<td>create.MK</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Sequential knockoff generator for genetic data
</h2>

<h3>Description</h3>

<p>Generate single/multiple knockoffs for genetic variants for customized analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">create.MK(X,pos,M=5,corr_max=0.75,maxN.neighbor=Inf,maxBP.neighbor=100000,
n.AL=floor(10*nrow(X)^(1/3)*log(nrow(X))),thres.ultrarare=25,
R2.thres=1,method='shrinkage',bigmemory=T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>

<p>A n*p genotype matrix, where n is the sample size and p is the number of genetic variants.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>

<p>A vector of length p. Location of the p genetic variants.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>M</code></td>
<td>

<p>Number of knockoffs per variant. The default is 5.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>corr_max</code></td>
<td>

<p>The correlation threshold for hierarchical clustering, such that variants from two different clusters do not have a correlation greater than corr_max. The hierarchical clustering step is a practical strategy to improve the power for tightly linked variants. The default is 0.75.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxN.neighbor</code></td>
<td>

<p>The maximum number of neighoring variables used to generate knockoffs. The default is Inf. Smaller number will inprove the computational efficiency, but the knockoffs will be less accurate.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxBP.neighbor</code></td>
<td>

<p>The size of neighboring region (in base pairs) used to generate knockoffs. The default is 100000.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.AL</code></td>
<td>

<p>The sample size for the algorithmic leveraging. The default is 10*n^(1/3)*log(n)).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thres.ultrarare</code></td>
<td>

<p>The minor allele count threshold that defines ultrarare variants. The knockoff generation for variants with minor allele counts below the threshold will be based on permutaton. The default is 25.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>R2.thres</code></td>
<td>

<p>The maximum R2 allowed in the auto-regressive model. More liberal values (&lt;1) lead to higher power for tightly linked variants, but the knockoffs will be less accurate.  The default is 1.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>

<p>The method for subsampling. The default is "shrinkage", corresponding to "shrinkage algorithmic leveraging".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bigmemory</code></td>
<td>

<p>Whether "bigmemory" operation is applied. Default is TRUE.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table><tr style="vertical-align: top;">
<td><code>X_k</code></td>
<td>

<p>An M dimentions list, where each dimention is an n*p matrix as a knockoff copy of original data.
</p>
</td>
</tr></table>
<h3>Examples</h3>

<pre><code class="language-R">
library(KnockoffScreen)

# load example vcf file from package "seqminer"
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])

# filter out constant variants
s&lt;-apply(example.G,2,sd)
example.G&lt;-example.G[,s!=0]
pos&lt;-as.numeric(gsub("^.*:","",colnames(example.G)))

# generate multiple knockoffs
example.G_k&lt;-create.MK(example.G,pos,M=5,corr_max=0.75)

</code></pre>


</div>