<div class="container">

<table style="width: 100%;"><tr>
<td>KSEA.Complete</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The Complete KSEA App Analysis</h2>

<h3>Description</h3>

<p>Takes a formatted phoshoproteomics data input and performs KSEA calculations to infer relative kinase activities
</p>


<h3>Usage</h3>

<pre><code class="language-R">KSEA.Complete(KSData, PX, NetworKIN, NetworKIN.cutoff, m.cutoff, p.cutoff)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>KSData</code></td>
<td>
<p>the Kinase-Substrate dataset uploaded from the file 
prefaced with "PSP&amp;NetworKIN_"
available from github.com/casecpb/KSEA/</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PX</code></td>
<td>
<p>the experimental data file formatted exactly as described below;
must have 6 columns in the exact order: Protein, Gene, Peptide, Residue.Both, p, FC;
cannot have NA values, or else the entire peptide row is deleted;
Description of each column in PX: 
</p>

<ul>
<li>
<p>"Protein" the Uniprot ID for the parent protein 
</p>
</li>
<li>
<p>"Gene" the HUGO gene name for the parent protein 
</p>
</li>
<li>
<p>"Peptide" the peptide sequence
</p>
</li>
<li>
<p>"Residue.Both" all phosphosites from that peptide, separated by semicolons if applicable;
must be formatted as the single amino acid abbrev. with the residue position (e.g. S102)
</p>
</li>
<li>
<p>"p" the p-value of that peptide (if none calculated, please write "NULL", cannot be NA)
</p>
</li>
<li>
<p>"FC" the fold change (not log-transformed); usually the control sample is the denominator
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NetworKIN</code></td>
<td>
<p>a binary input of TRUE or FALSE, indicating whether or not to include NetworKIN predictions; 
NetworKIN = TRUE means inclusion of NetworKIN predictions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NetworKIN.cutoff</code></td>
<td>
<p>a numeric value between 1 and infinity setting the minimum NetworKIN score 
(can be left out if NetworKIN = FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m.cutoff</code></td>
<td>
<p>a numeric value between 0 and infinity indicating the min. # of substrates 
a kinase must have to be included in the bar plot output</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.cutoff</code></td>
<td>
<p>a numeric value between 0 and 1 indicating the p-value cutoff for
indicating significant kinases in the bar plot</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>creates the following outputs that are deposited into your working directory: 
a bar plot highlighting key kinase results, a .csv file of all KSEA kinase scores, 
and a .csv file listing all kinase-substrate relationships used for the calculations
</p>


<h3>References</h3>

<p>Casado et al. (2013) Sci Signal. 6(268):rs6
</p>
<p>Hornbeck et al. (2015) Nucleic Acids Res. 43:D512-20
</p>
<p>Horn et al. (2014) Nature Methods 11(6):603-4
</p>


<h3>Examples</h3>

<pre><code class="language-R">KSEA.Complete(KSData, PX, NetworKIN=TRUE, NetworKIN.cutoff=5, m.cutoff=5, p.cutoff=0.01)
KSEA.Complete(KSData, PX, NetworKIN=FALSE, m.cutoff=2, p.cutoff=0.05)

</code></pre>


</div>